airflow/utils/db.py

```diff
@@ -64,13 +65,19 @@ def wrapper(*args, **kwargs):
         session_in_args = arg_session in func_params and \
             func_params.index(arg_session) < len(args)
         session_in_kwargs = arg_session in kwargs
-
-        if session_in_kwargs or session_in_args:
-            return func(*args, **kwargs)
-        else:
-            with create_session() as session:
-                kwargs[arg_session] = session
-                return func(*args, **kwargs)
+        try_number = 1
+        while try_number < max_tries:
+            try:
+                if session_in_kwargs or session_in_args:
+                    return func(*args, **kwargs)
+                else:
+                    with create_session() as session:
+                        kwargs[arg_session] = session
+                        return func(*args, **kwargs)
+            except Exception as e:
+                log.warn(f'Try {try_number}/{max_tries} failed to perform db action.', e)
+                time.sleep(min(30., .01 * (1 << try_number)))
```

Comment:
.01 * (1 << 5) = .32 seconds?

Reply:
max_tries is set by the caller; max timeout never gets set, so cap it to 30s

Reply:
but if we're doing exponential backoff, then a .32 second sleep after 4 tries seems like it's not what we want? maybe

Reply:
nope, we want 0.1, not 0.01, typo'd the decimal